yo/app/scripts/directives/transformation.js

```diff
@@ -1132,25 +1132,44 @@ angular.module('dmpApp')
         }
 
         /**
-         * Send a single transformation to the server
+         * This actually sends a Transformation
+         * @param {boolean} persist - Should the data be persisted or just previewed
          */
-        $scope.sendTransformation = function() {
+        function sendTransformationsInternal(persist) {
```

Comment:
ok, I made it a bit more clear via parametrizing the mappings for the job of the task.